ESP32: Update README of esp32 supported examples (#7225)
project-chip/connectedhomeip: examples/all-clusters-app/esp32/README.md

# CHIP All Clusters Example

A prototype application that demonstrates device commissioning and cluster
control.

---

-   [CHIP All Clusters Example](#chip-all-clusters-example)
    -   [Supported Devices](#supported-devices)
    -   [Building the Example Application](#building-the-example-application)
        -   [To build the application, follow these steps:](#to-build-the-application-follow-these-steps)
    -   [Commissioning and cluster control](#commissioning-and-cluster-control)

>         -   [Setting up python controller](#setting-up-python-controller)

Reviewer: Can we remove `To build the application, follow these steps:` from being a separate section? I don't think that a separate heading is required for that

Author: Done

>         -   [Commissioning esp32 device](#commissioning-using-ble)

Reviewer: esp32 => ESP32 (at all places)

        -   [Cluster control](#cluster-control)
    -   [Note](#note)

---

## Supported Devices

The CHIP demo application is intended to work on three categories of ESP32
devices: the
[ESP32-DevKitC](https://www.espressif.com/en/products/hardware/esp32-devkitc/overview),
the
[ESP32-WROVER-KIT_V4.1](https://www.espressif.com/en/products/hardware/esp-wrover-kit/overview),

> and the [M5Stack](http://m5stack.com).

Reviewer: Explicitly mention M5Stack Core. As M5Stack Core 2 is not supported yet.

-   Setting up the environment

        $ cd ${HOME}/tools/esp-idf
        $ ./install.sh
        $ . ./export.sh
        $ cd {path-to-connectedhomeip}

    To download and install packages.

        $ source ./scripts/bootstrap.sh
        $ source ./scripts/activate.sh

>     If packages are already installed then simply activate it.

Reviewer 2: ```suggestion
    If packages are already installed then simply activate them.
```

Author: Done.

        $ source ./scripts/activate.sh

-   Configuration Options

> To choose from the different configuration options, run menuconfig

Reviewer 2: ```suggestion
To choose from the different configuration options, run menuconfig.
```

Author: Done

          $ idf.py monitor ESPPORT=/dev/tty.SLAB_USBtoUART

## Commissioning and cluster control

> Commissioning can be carried out using WiFi, BLE or ByPass.

Reviewer 2: ```suggestion
Commissioning can be carried out using WiFi, BLE or Bypass.
```

> 1.  Set the `Rendezvous Mode` for commissioning using menuconfig, the default

Reviewer 2: ```suggestion
1.  Set the `Rendezvous Mode` for commissioning using menuconfig; the default
```

    Rendezvous mode is BLE.

         $ idf.py menuconfig

Select the Rendezvous Mode via `Demo -> Rendezvous Mode`. If Rendezvous Mode is

> ByPass then set the credentials of WiFi Network i.e SSID and Password from
> menuconfig.

Reviewer 2: ```suggestion
Bypass then set the credentials of the WiFi Network (i.e. SSID and Password from
menuconfig).
```

`idf.py menuconfig -> Component config -> CHIP Device Layer -> WiFi Station Options`

2.  Now flash the device with the same command as before. (Use the right `/dev`
    device)

          $ idf.py flash monitor ESPPORT=/dev/tty.SLAB_USBtoUART

> 3.  The device should boot up and when device connects to your network then you
>     will see a log like this in the monitor.

Reviewer 2: ```suggestion
3.  The device should boot up.  When the device connects to your network, you
    will see a log like this in the monitor.
```

          I (5524) chip[DL]: SYSTEM_EVENT_STA_GOT_IP
          I (5524) chip[DL]: IPv4 address changed on WiFi station interface: <IP_ADDRESS>...

> 4.  Controller used

Reviewer 3: ```suggestion
4.  Use
```

    [python based device controller](https://github.com/project-chip/connectedhomeip/tree/master/src/controller/python)
    or
    [standalone chip-tool](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool)
    or
    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin/CHIPTool)
    or
    [Android chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool)
    to communicate with the device.

Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
your network configuration. To erase it, simply run.

          $ idf.py erase_flash ESPPORT=/dev/tty.SLAB_USBtoUART

### Setting up python controller

> Once ESP32 is up and running, we need to setup device controller to perform

Reviewer 3: ```suggestion
Once ESP32 is up and running, we need to set up a device controller to perform
```

commissioning and cluster control.

-   Set up python controller

           $ cd {path-to-connectedhomeip}
           $ ./scripts/build_python.sh -m platform

-   Execute the controller.

           $ source ./out/python_env/bin/activate

>            $ sudo chip-device-ctrl

Reviewer 3: This sudo bit is ... not desirable.  Why are we asking people to do that?

Author: Removed.

          $ idf.py monitor ESPPORT=/dev/tty.SLAB_USBtoUART

## Commissioning and cluster control

Commissioning can be carried out using WiFi, BLE or Bypass.

1.  Set the `Rendezvous Mode` for commissioning using menuconfig; the default
    Rendezvous mode is BLE.

         $ idf.py menuconfig

Select the Rendezvous Mode via `Demo -> Rendezvous Mode`. If Rendezvous Mode is
ByPass then set the credentials of the WiFi Network (i.e. SSID and Password from
menuconfig).

`idf.py menuconfig -> Component config -> CHIP Device Layer -> WiFi Station Options`

2.  Now flash the device with the same command as before. (Use the right `/dev`
    device)

          $ idf.py flash monitor ESPPORT=/dev/tty.SLAB_USBtoUART

3.  The device should boot up. When device connects to your network, you will

>     see a log like this in the monitor.

Reviewer: in the monitor => on the device console?

          I (5524) chip[DL]: SYSTEM_EVENT_STA_GOT_IP
          I (5524) chip[DL]: IPv4 address changed on WiFi station interface: <IP_ADDRESS>...

4.  Use
    [python based device controller](https://github.com/project-chip/connectedhomeip/tree/master/src/controller/python)
    or
    [standalone chip-tool](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool)
    or
    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin/CHIPTool)
    or
    [Android chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool)
    to communicate with the device.

Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
your network configuration. To erase it, simply run.

          $ idf.py erase_flash ESPPORT=/dev/tty.SLAB_USBtoUART

> ### Setting up python controller

Reviewer: python controller -> Python Controller

            The same value should be used in the next commands.
            We have chosen a random node ID which is 135246.

-   Add credentials of the Wi-Fi network you want the ESP32 to connect to, using
    the `AddWiFiNetwork` command and then enable the ESP32 to connect to it
    using `EnableWiFiNetwork` command. In this example, we have used `TESTSSID`
    and `TESTPASSWD` as the SSID and passphrase respectively.

         - chip-device-ctrl > zcl NetworkCommissioning AddWiFiNetwork 135246 0 0 ssid=str:TESTSSID credentials=str:TESTPASSWD breadcrumb=0 timeoutMs=1000

         - chip-device-ctrl > zcl NetworkCommissioning EnableNetwork 135246 0 0 networkID=str:TESTSSID breadcrumb=0 timeoutMs=1000

-   Close the BLE connection to ESP32, as it is not required hereafter.

         - chip-device-ctrl > close-ble

-   Resolve DNS-SD name and update address of the node in the device controller.

         - chip-device-ctrl > resolve 0 135246

### Cluster control

-   After successful commissioning, use the OnOff cluster command to control the
    OnOff attribute. This allows you to toggle a parameter implemented by the
    device to be On or Off.

>     `chip-device-ctrl > zcl OnOff Off 135246 0 0`

Reviewer: `0 0` => `1 0` for cluster commands (at all the places)

### Note

> This demo app illustrated controlling OnOff cluster (Server) attributes of an

Reviewer: illustrates?
